### PR TITLE
feat: dotnet new templates checkups and solution

### PR DESCRIPTION
**Pull request by MartinZikmund**
Checks for the latest version of Uno and Extensions templates and installs them if available.

### CODE REVIEW

File: UnoCheck/Checkups/DotNetNewTemplatesCheckupBase.cs [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+        }
+
+        return DiagnosticResult.Ok(this);
+    }
+
+    private string GetDotNetNewInstalledList()
+    {
+        // Running 'dotnet new uninstall' without any package ID will list all installed
+        // dotnet new templates along with their versions.
+        var processInfo = new ProcessStartInfo("dotnet", "new uninstall");
```
Comment (Youssef1313) [inline UnoCheck/Checkups/DotNetNewTemplatesCheckupBase.cs:58]: A more robust approach would be using `Microsoft.TemplateEngine.Edge` NuGet package. I haven't used it before, but I believe this is what the .NET CLI uses underneath the hoods.

That way we don't have to worry about .NET CLI changing the format of the produced output.

**Comment by jeromelaban** [comment]
We need to adjust this for the new nuget package template name which is not available yet.

File: UnoCheck/Checkups/DotNetNewTemplatesCheckupBase.cs [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+using DotNetCheck.Solutions;
+using NuGet.Versioning;
+using System.Diagnostics;
+using System.Linq;
+using System.Text.RegularExpressions;
+using System.Threading.Tasks;
+
+namespace DotNetCheck.Checkups;
+
+internal abstract class DotNetNewTemplatesCheckupBase : Checkup
```
Comment (Youssef1313) [inline UnoCheck/Checkups/DotNetNewTemplatesCheckupBase.cs:14]: Do we still need this base class now that we have a single derived class?

**Comment by MartinZikmund** [comment]
TODO: Add uninstall `dotnet new uninstall Uno.ProjectTemplates.Dotnet`

File: UnoCheck/Checkups/DotNetNewTemplatesCheckupBase.cs (hunk 1 above)
Comment (MartinZikmund) [reply UnoCheck/Checkups/DotNetNewTemplatesCheckupBase.cs]: Tried to use the library, but I wasn't able to get it work and there is not many useful samples for it. So for now, keeping the original approach, but if you have some resources, I would look into it further 👍

**Review by jeromelaban — APPROVED** [review]

**Pull request merged**